### PR TITLE
Adding GitHub actions and API

### PR DESCRIPTION
**Pull request by sayakpaul**

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      string: ${{ steps.changestring.outputs.string}}
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: google-github-actions/deploy-appengine@v0.1.0
+      - name: Set up AppEngine deployer.
+        with:
+          project: ${{ secrets.gcp_project }}
+          deliverables: app.yaml
+          credentials: ${{ secrets.gcp_credentials }}
```
Comment (deep-diver) [inline .github/workflows/deploy.yml:22]: I think `secrets.gcp_project` and `secrets.gcp_credentials` are not set yet?

File: app.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+runtime: python39 # or another supported version
+
+instance_class: F4_1G
```
Comment (deep-diver) [inline app.yaml:3]: I don't know much about GCP App Engine, but don't we need `entrypoint`?

File: server/api.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+def get_images():
+    tag = request.get("t")
+    query = request.get("s_query")
+    top_k = request.get("k")
+
+    results = searcher.get_similar_images(tag, query, top_k)
+    encoded_imges = []
+
+    for result in results:
+        encoded_imges.append(get_response_image(result))
```
Comment (deep-diver) [inline server/api.py]: I am sorry to say this at this moment.

It seems like sending entire images in a byte array is too much overhead. Can we just retrieve back the image URL (Pixabay) from the model output?

=> **please check out the PR #6 **

**Review by deep-diver — CHANGES_REQUESTED** [review]

**Comment by deep-diver** [comment]
@sayakpaul 
can you please look into some of my comments?

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (sayakpaul) [reply .github/workflows/deploy.yml:22]: Right. I think you can only set that as the repo owner?

File: app.yaml (hunk 2 above)
Comment (sayakpaul) [reply app.yaml:3]: That is handled here:

```yaml
handlers:
  - url: /search
```

You can check [this link](https://cloud.google.com/appengine/docs/standard/python3/building-app/writing-web-service) out to quickly get an overview of AppEngine.

File: server/api.py (hunk 3 above)
Comment (sayakpaul) [reply server/api.py]: Sure. I will take care of that :)

**Comment by sayakpaul** [comment]
@deep-diver I have addressed your comments. Please take a look when you get a moment.

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (deep-diver) [reply .github/workflows/deploy.yml:22]: oh ok. For an experimental purpose, let me setup the environment variables taken from my GCP free account (w/ $300).

**Review by deep-diver — APPROVED** [review]

**Pull request merged**